api_volontaria/apps/volunteer/models.py

```diff
@@ -325,12 +327,45 @@ def send_email_confirmation(self):
             },
         }
 
-        EmailAPI().send_template_email(
-            self.user.email,
-            'CONFIRMATION_PARTICIPATION',
-            context,
-        )
-
+        # Checking whether email template already exists;
+        # If not, use default template
+        TEMPLATES = settings.ANYMAIL.get('TEMPLATES')
+        id = TEMPLATES.get('CONFIRMATION_PARTICIPATION')
+        if id:
+            EmailAPI().send_template_email(
+                self.user.email,
+                'CONFIRMATION_PARTICIPATION',
+                context,
+            )
+        else:
+            merge_data = {
+                'TYPE_PARTICIPATION': type_participation,
+                'FIRST_NAME' : self.user.first_name,
+                'EVENT_NAME' : self.event.description,
+                'TASK_TYPE' : self.event.task_type.name,
+                'START_DATE' : context['ACTIVITY']['START_DATE'],
+                'START_TIME' : context['ACTIVITY']['START_TIME'],
+                'END_TIME' : context['ACTIVITY']['END_TIME'],
+                'LOCATION_NAME': self.event.cell.name,
+                'ADDRESS_LINE_1': self.event.cell.address_line_1,
+                'ADDRESS_LINE_2': self.event.cell.address_line_2,
+                'POSTAL_CODE': self.event.cell.postal_code,
+                'CITY': self.event.cell.city,
+                'STATE_PROVINCE': self.event.cell.state_province,
```

Comment:
When you use the variable in a Django template you can directly use objects like that, it help you avoid creating too much variable for nothing:

**models.py**
```python
[...]

merge_data={
    'CELL': self.event.cell
}

[...]
```

**my_template.html**
```html
[...]

<p>{{ cell.name }}</p>
<p>{{ cell.address_line_1 }}</p>

<!-- Here we can use a condition since this field is optional -->
{% cell.address_line_2 %}
<p>{{ cell.address_line_2 }}</p>
{% cell.address_line_2 %}

[...]
```

---

But in this context maybe it's better to use the already existing context for our intern emails, so we just have one dictionnary to maintain in the future if some refactor are needed.